cluster/etcd_service_discovery.go

```diff
@@ -530,3 +631,12 @@ func (sd *etcdServiceDiscovery) watchEtcdChanges() {
 		}
 	}(w)
 }
+
+func (sd *etcdServiceDiscovery) isServerTypeBlacklisted(svType string) bool {
+	for _, blacklistedSv := range sd.serverTypesBlacklist {
```

Comment:
Isn't this going to crash if sd.serverTypesBlacklist is nil? (I saw that it's the default value in the config)

Reply:
No, a nil slice is considered an empty slice in go.
https://play.golang.org/p/GgbP0zUX8kP